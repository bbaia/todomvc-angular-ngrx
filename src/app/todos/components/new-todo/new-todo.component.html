<input
  id="new-todo"
  class="new-todo"
  type="text"
  autofocus
  placeholder="What needs to be done?"
  #textInput
  (keyup.enter)="newTodo(textInput.value); textInput.value = ''"
/>
