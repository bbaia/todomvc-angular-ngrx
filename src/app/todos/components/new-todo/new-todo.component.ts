import { Component, EventEmitter, Output } from '@angular/core';

@Component({
  selector: 'app-new-todo',
  templateUrl: './new-todo.component.html',
})
export class NewTodoComponent {
  @Output() addTodo = new EventEmitter<string>();

  newTodo(text: string): void {
    if (text && text.trim()) {
      this.addTodo.emit(text.trim());
    }
  }
}
