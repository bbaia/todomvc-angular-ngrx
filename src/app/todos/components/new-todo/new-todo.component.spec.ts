import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NewTodoComponent } from './new-todo.component';

describe('NewTodoComponent', () => {
  let component: NewTodoComponent;
  let fixture: ComponentFixture<NewTodoComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [NewTodoComponent],
    });
    fixture = TestBed.createComponent(NewTodoComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeDefined();
  });

  it('should render', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should raise "addTodo" event', done => {
    fixture.detectChanges();

    component.addTodo.subscribe((text: string) => {
      // should trim text
      expect(text).toBe('Test component!');
      done();
    });
    const input: HTMLInputElement =
      fixture.debugElement.nativeElement.querySelector('.new-todo');
    input.value = '  Test component!  ';
    input.dispatchEvent(new Event('input'));
    input.dispatchEvent(new KeyboardEvent('keyup', { key: 'Enter' }));

    // should clear input field after raising event
    expect(
      fixture.debugElement.nativeElement.querySelector('.new-todo').value,
    ).toEqual('');
  });
});
