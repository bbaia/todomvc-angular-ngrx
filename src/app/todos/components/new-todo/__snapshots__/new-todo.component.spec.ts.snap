// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoComponent should render 1`] = `
<app-new-todo
  __ngContext__={[Function LRootView]}
  addTodo={[Function EventEmitter_]}
>
  <input
    autofocus=""
    class="new-todo"
    id="new-todo"
    placeholder="What needs to be done?"
    type="text"
  />
</app-new-todo>
`;
